app/controllers/application_controller.rb

```diff
@@ -1,2 +1,26 @@
 class ApplicationController < ActionController::API
+  private
+
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+
+  def set_the_current_tenant
+    return unless ENV["ENFORCE_TENANCY"]
+
+    tenant = Tenant.find_or_create_by(:external_tenant => identity_account_number) if identity_account_number
```

Comment:
Suggest moving `identity_account_number` into a variable so we do not have to run the method twice.  Or cache it in the method if you think there is any value in that.  But I feel other callers could get the value from checking the current tenant.